src/middlewares/existingSourceMiddleware.js

```diff
@@ -0,0 +1,16 @@
+const sources = require('../data/sources.json');
+const path = require('path');
+const eSources = sources.map((source) => source.name); //Existing sources
```

Comment:
The commented explanation is a good idea, but you could be even more explicit by naming the constant `existingSources`, I didn't realise the significance until I read the whole thing through a few times.

I only say this to keep it crystal clear what is happening here (which is very nice btw!)

Reply:
You're right, should've kept it simple. I'll change it right away.